component/service/src/main/java/org/exoplatform/social/rest/impl/activity/ActivityRestResourcesV1.java

```diff
@@ -1076,7 +1076,7 @@ private List<DataEntity> convertToEntities(List<ExoSocialActivity> activities,
   private void addPreloadActivityIds(List<String> activityIds, int preloadLimit, String expand, ResponseBuilder responseBuilder) {
     int offset = preloadLimit > MAX_TO_PRELOAD ? preloadLimit - MAX_TO_PRELOAD : 0;
     List<String> preloadActivityIds =
-                                    activityIds.size() >= preloadLimit ? activityIds.subList(offset, preloadLimit) : activityIds;
+                                    activityIds.size() >= preloadLimit ? activityIds.subList(offset, preloadLimit) :  activityIds.subList(offset > MAX_TO_PRELOAD ? offset : 0, activityIds.size());
```

Comment:
recalculate the preload limit the salme as the offset in a dedicated statement to make the code more readable

Reply:
done !